sdk/nodejs/resource.ts

```diff
@@ -325,6 +319,23 @@ export abstract class Resource {
         return utils.isInstance<Resource>(obj, "__pulumiResource");
     }
 
+    private static callsites(): any[] {
+        const _prepareStackTrace = Error.prepareStackTrace;
+        try {
+            let callsites: any[] = [];
+            Error.prepareStackTrace = (_, stack) => {
+                callsites = stack.slice(1);
```

Comment:
I'm curious: For TypeScript code, does this reference the transpiled code or the original source position (via source maps)? 

I suppose "it depends" on how you're running the TS code, e.g. whether you're using the built-in ts-node support, pre-transpiling, specifying `"sourceMap": true` in tsconfig.json, specifying `--enable-source-maps` in Pulumi.yaml's `nodeargs` runtime options, etc.

Reply:
That's a great question--I'll double-check. It could be that the mapping to original positions is happening in the default implementation of `prepareStackTrace`, in which case these changes would lose the mapping.

Reply:
Okay it does look like the mapping is happening inside of the default `prepareStackTrace`, so this will lose original positions. Switched to a different implementation that parses stringified stacks into frames.